app/controllers/menu_controller.rb

```diff
@@ -1,19 +1,77 @@
 class MenuController < ApplicationController
-    def menu
-        # display the menu for the current day
-        date = Time.now.strftime("%d/%m/%Y")
-        dishes = TemporaryMenu.get_dishes_by_date(date)
-        @dishes = dishes.to_a
-    end
+    skip_before_action :verify_authenticity_token
     
-    def new_date
-        # byebug
-        # One of our hardworking team member spent one week on this bug 
-        # and we think it is worth commemorating
-        date = params[:date]
-        dishes = TemporaryMenu.get_dishes_by_date(date)
+    #finds the dishes and their recipies for a given meal_type and given day
+    # spits out an array in the format [{name:"kukuname",ingredients:[ingredient:"kukuingredient",portion_size:"kukulargeportion"]}]
+    def find_dishes(meal_type,day)
+        dishes_arr = Menu.get_dishes_by_day_type(day,meal_type)
+        dishes = []
+        dishes_arr.each do |dish_menu|
+            dish = Dish.where(id:dish_menu.dish_id).first
+            if dish != nil 
+                dish_recipies = Recipe.get_recipe_by_dish(dish)
+                ingredients = []
+                dish_recipies.each do |recipe|
+                    ingredients.push(OpenStruct.new({"ingredient" => Ingredient.where(id:recipe.ingredient_id).first.name, "portion_size" => recipe.portion_size}) )
+                end 
+                dishes.push(OpenStruct.new({"name" => dish.name, "ingredients" => ingredients}) )
+            end
+        end 
+        return dishes
+    end 
+
+    def menu
+        # 1'st day hardcoded for now that the db and seed doesn't have all the data. Later, well use date
+        date = params[:date] || Time.now.strftime("%d/%m/%Y")
+        @lunch_dishes = find_dishes("Dinner",1)
+        puts @lunch_dishes
+    end 
 
-        @dishes = dishes.to_a
+
+    # edit the menu (add/remove dish)
+    # week, day are url params (see routes)
+    # request.body requires: 
+    #   - change_action: add OR remove
+    #   - dish_name: name of the dish
+    #   - type: type of the meal
+    # if things go wrong, spits out a flash message
+   def edit
+    week = params[:week].to_i
+    day = params[:day].to_i
+    error = nil
+    if week > 7 || week < 0 || day > 31 || day < 0
+        error = 'Impossible week/day url parameters'
+    else 
+        begin
+            action = params[:change_action]
+            dish_name = params[:dish_name]
+            type = params[:type]
+            dish = Dish.where(name:dish_name).first
+            day = 7*(week-1) + day
```

Comment:
Would be correct if 1 <= day <= 7

Reply:
Kuku! So what is day in our db? I assumed that it's like "date" (between 1 and 31) cause menu doesn't have ween but only day, and I know that we have 49 days in a cycle.

Reply:
The model is assuming it's between 1-49. So, 7 weeks of 7 days should work, or you can ask for raw number between 1-49.

Reply:
Bogdan, did you just specify that if a day is bigger than 1 or smaller than 7, it is an invalid day?!!!!!!!!!!!!!!!!!!!
Congrats, you made it error on all possible inputs. :|

Reply:
Also, the final day is supposed to be in the 1-49 range. So, therefore, week should be in the 0-6 range and day should also be in the 0-6 range, except when you get the final day, you should add 1 to it. Or I can go change the models and Garrett's seed to fix it.